### PR TITLE
Update versioning and image repositories

### PR DESCRIPTION
**Pull request by mangelajo**
Update helm charts to follow the submariner versioning,
and include the image repositories by default.

### CODE REVIEW

File: submariner/Chart.yaml [hunk 1]
```diff
@@ -1,8 +1,8 @@
 ---
 name: submariner
-version: 0.0.4
-appVersion: v0.0.4
-description: k8s 1.16.x support
+version: 0.4.0-rc3
+appVersion: 0.4.0-rc3
+description: Libreswan support
```
Comment (mkolesnik) [inline submariner/Chart.yaml:5]: Shouldn't this be describing submariner?

File: submariner/templates/engine-deploy.yaml [hunk 2]
```diff
@@ -108,7 +108,7 @@ spec:
           value: "{{ .Values.leadership.renewDeadline }}"
         - name: LEADERSHIP_RETRYPERIOD
           value: "{{ .Values.leadership.retryPeriod }}"
-        image: {{ .Values.engine.image.repository }}:{{ .Values.engine.image.tag }}
+        image: {{ .Values.engine.image.repository }}:{{ default .Chart.AppVersion .Values.engine.image.tag }}
```
Comment (mkolesnik) [inline submariner/templates/engine-deploy.yaml:111]: Hope you checked this doesn't break the local repository feature in shipyard :)

File: submariner/Chart.yaml (hunk 1 above)
Comment (mangelajo) [reply submariner/Chart.yaml:5]: Ah, I thought it was the description of the version... more than the chart itself, you're probably right.

File: submariner/templates/engine-deploy.yaml (hunk 2 above)
Comment (mangelajo) [reply submariner/templates/engine-deploy.yaml:111]: I didn't find a way of doing it, may be we need to add some E2E CI to this repo too. But .Values.engine.image.tag will always override .Chart.AppVersion acording to helm documentation.

File: submariner/Chart.yaml (hunk 1 above)
Comment (mangelajo) [reply submariner/Chart.yaml:5]:
```suggestion
description: Submariner enables direct networking between Pods and Services in different Kubernetes clusters
```

File: submariner-k8s-broker/Chart.yaml [hunk 3]
```diff
@@ -1,7 +1,7 @@
 ---
 name: submariner-k8s-broker
-version: 0.0.2
-appVersion: v0.0.2
+version: 0.4.0-rc3
+appVersion: 0.4.0-rc3
```
Comment (mangelajo) [inline submariner-k8s-broker/Chart.yaml:4]:
```suggestion
version: 0.4.0
appVersion: 0.4.0
```

File: submariner/Chart.yaml [hunk 4]
```diff
@@ -1,8 +1,8 @@
 ---
 name: submariner
-version: 0.0.4
-appVersion: v0.0.4
-description: k8s 1.16.x support
+version: 0.4.0-rc3
+appVersion: 0.4.0-rc3
```
Comment (mangelajo) [inline submariner/Chart.yaml:4]:
```suggestion
version: 0.4.0
appVersion: 0.4.0
```

**Review by mkolesnik — APPROVED** [review]
LGTM

**Review by tpantelis — APPROVED** [review]

**Pull request merged**